src/test_server.py

```diff
@@ -1,20 +1,48 @@
 # -*- coding: utf-8 -*-
-"""asdfa"""
+"""Test for the client server pair for echo."""
 from __future__ import unicode_literals
 import pytest
 
 
-SERVER_PARAMS_TABLE = [
-    ('Hello', 'Hello'),
+SERVER_OK_PARAMS = [
+    ('Hello', 'HTTP/1.1 200 OK\r\n'),
     ('hello my name is chris and i am in python 401',
-     'hello my name is chris and i am in python 401'),
-    ('12345678', '12345678'),
-    ('1234567890123456', '1234567890123456'),
-    ('««««', '««««')
+     'HTTP/1.1 200 OK\r\n'),
+    ('12345678',
+     'HTTP/1.1 200 OK\r\n'),
 ]
 
 
-@pytest.mark.parametrize('message, result', SERVER_PARAMS_TABLE)
-def test_client_server_echo(message, result):
+ERROR_PARAMS = [
+    (b'HTTP/1.1 500 Internal Server Error\r\n\r\n')
+]
+
+
+OK_PARAMS = [
+    (b'HTTP/1.1 200 OK\r\n')
+]
+
+
+@pytest.mark.parametrize('message, result', SERVER_OK_PARAMS)
+def test_client_server_response_ok(message, result):
+    """Test message send and recieve."""
     from client import client
-    assert client(message) == result
+    message = client(message)
+    message = message[0:-39]
+    assert message == result
+
+
+@pytest.mark.parametrize('result', ERROR_PARAMS)
+def test_response_error(result):
+    """Test message send and recieve."""
+    from server import response_error
+    assert response_error() == result
+
```

Comment:
It's good you're testing the response error function, but don't forget to test what happens if you actually send a bad request.